hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/HttpServer2Metrics.java

```diff
@@ -91,4 +90,10 @@ public void getMetrics(MetricsCollector collector, boolean all) {
         .addGauge(HttpServer2MetricsInfo.HttpServerThreadQueueWaitingTaskCount,
             threadPool.getQueueSize());
   }
+
+  public void unRegister() {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    ms.unregisterSource(NAME);
```

Comment:
If we want to allow multiple `HttpServer2` objects to share the same `HttpServer2Metrics` instance, then it should be reference-counted, and unregistration should happen only when the last reference is removed.

Reply:
I'm not aware of it.  But then why bother with making it a singleton?

Reply:
Initially to fix the test, but , i think having `HttpServer2Metrics` `unRegister` when `HttpServer2` stops will fix the test, needn't make it a singleton